src/app/dashboard/whitelist/WhiteListPage.js

```diff
@@ -0,0 +1,378 @@
+  TableToolbarSearch, TableToolbarContent } = DataTable
+
+type StateProps = {|
+  transferManager: TransferManager,
+  addresses: Array<string>,
+  sell: Array<number>,
+  buy: Array<number>,
+  investors: Array<Investor>,
+  csvMessage: string,
+  previewCSVShowing: boolean,
+  listLength: number,
+|}
+
+type DispatchProps = {|
+  initialize: () => any,
+  handleUpload: () => any,
+  multiSubmit: () => any,
+  singleSubmit: () => any,
+  getWhitelist: (?Date, ?Date) => any,
+  updateListLength: (number) => any,
+  removeInvestor: (investors: Array<string>) => any,
+  editInvestors: (investors: Array<string>) => any,
+|}
+
+const mapStateToProps = (state) => ({
+  transferManager: state.whitelist.transferManager,
+  addresses: state.whitelist.addresses,
+  sell: state.whitelist.sell,
+  buy: state.whitelist.buy,
+  investors: state.whitelist.investors,
+  csvMessage: state.whitelist.csvMessage,
+  previewCSVShowing: state.whitelist.previewCSVShowing,
+  listLength: state.whitelist.listLength,
+})
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  initialize: () => dispatch(initialize()),
+  handleUpload: (htmlCsv) => dispatch(uploadCSV(htmlCsv.target.files[0])),
+  multiSubmit: () => dispatch(multiUserSubmit()),
+  singleSubmit: () => dispatch(oneUserSubmit()),
+  getWhitelist: (calenderStart: Date, calenderEnd: Date) => dispatch(getWhitelist(calenderStart, calenderEnd)),
+  updateListLength: (pageNumber: number) => dispatch(listLength(pageNumber)),
+  removeInvestor: (investors: Array<string>) => dispatch(removeInvestor(investors)),
+  editInvestors: (investors: Array<string>) => dispatch(editInvestors(investors)),
+})
+
+type Props = StateProps & DispatchProps
+
+type State = {
+  page: number,
+  editInvestorsShowing: boolean,
+  editInvestors: Array<string>,
+}
+
+ type EventData = {
+  id: string,
+  address: string,
+  added: ?string,
+  addedBy: ?string,
+  from: string,
+  to: string,
+}
+
+type PageChanger = {
+  page: number,
+  pageSize: number,
+}
+
+type DatePickerType = [Date, Date]
+
+const dateFormat = (date: Date) => date.toLocaleDateString('en', {
+  year: 'numeric',
+  month: 'short',
+  day: 'numeric',
+})
+
+class WhitelistPage extends Component<Props, State> {
+  state = {
+    page: 0,
+    editInvestorsShowing: false,
+    editInvestors: [],
+  }
+
+  componentWillMount () {
+    this.props.initialize()
+  }
+
+  handleChangePages = (pc: PageChanger) => {
+    this.props.updateListLength(pc.pageSize)
+    this.setState({
+      page: (pc.page - 1),
+    })
+  }
+
+  handleDatePicker = (picker: DatePickerType) => {
+    if (picker.length === 2){
+      this.setState({
+        page: 0, //reset to initial page , othewise it might refer to a page that doesnt exist //TODO: make sure this is true, im not sure it is
+      })
+      this.props.getWhitelist(picker[0],picker[1])
+    }
+  }
+
+  handleEditInvestors = (investors: Array<Object>) => {
```

Comment:
We have Investor type here instead of object

Reply:
This should have been named differently by me. It is actually an object that is returned from the carbon data component , and then from that object I extract the investor addresses. So I have renamed it to 'dataTableRow', instead of 'investors'